test/cqlpy/test_filtering.py

```diff
@@ -486,3 +487,17 @@ def test_selecting_a_regular_column_does_not_poison_filtering_of_a_static_row(cq
         cql.execute(f"INSERT INTO {table} (a, s) VALUES (1, 2)")
         res = cql.execute(f"SELECT a, b, c, s FROM {table} WHERE s = 2 ALLOW FILTERING")
         assert list(res) == [(1, None, None, 2)]
+
+# Test that some other CQL commands support the same filtering as SELECT, i.e. one based on the WHERE clause.
```

Comment:
You called this LWT "IF" filtering both here and in the commit message, and I think this is misleading. This is not "filtering". Also in DynamoDB, "FilterExpression" and "ConditionExpression" have the same syntax (called condition_expression in alternator/expressions.g) but we don't speak of "Filtering" (a Query/Scan feature) when it's a "Condition" (an Update/Delete feature) or vice versa, it's confusing.

This is also why I think this test doesn't belong in test_filtering.py.

Another reason why this PR is NOT about filtering: This PR is NOT about https://github.com/scylladb/scylladb/issues/10593 !